met-api/src/met_api/services/comment_service.py

```diff
@@ -73,11 +78,23 @@ def __form_comment(component_id, comment_text, survey_submission: SubmissionSche
             'submission_id': survey_submission.get('id', None)
         }
 
+    @classmethod
+    def extract_components(cls, survey_form: dict):
+        """Extract components from survey form."""
+        components = list(survey_form.get('components', []))
+
+        if survey_form.get('display') == cls.form_display:
+            return components
```

Comment:
So if we ever enable a panel or "container" components this won't work right?

Reply:
I wonder if our code should always check for iterable components

Reply:
No, this code does not cover that scenario, but it's not in scope for this task. This is a simple solution just to allow for the multi-page submission. To check for nested iterable components to allow usage of layout components in survey is more complex and better handled separately.